src/AzSK.ADO/Framework/Core/STMapping/AzSKADOServiceMapping.ps1

```diff
@@ -74,20 +78,92 @@ class AzSKADOServiceMapping: CommandBase
             }
+        }       
+
+        # Get Build-Repo mappings
+        try {            
+            $buildObjectListURL = ("https://dev.azure.com/{0}/{1}/_apis/build/definitions?queryOrder=lastModifiedDescending&api-version=6.0" +'&$top=10000') -f $($this.orgName), $this.projectName;       
+            $buildObjectList = $this.GetBuildReleaseObjects($buildObjectListURL,'Build');
+            $counter =0
+            foreach ($build in $buildObjectList) {               
+                try {                
+                    $counter++
+                    Write-Progress -Activity 'Build mappings...' -CurrentOperation $build.name -PercentComplete (($counter / $buildObjectList.count) * 100)
+                    $buildSTData = $this.BuildSTDetails.Data | Where-Object { ($_.buildDefinitionID -eq $build.id) };
+                    if(!($buildSTData))
+                    {
+                        $buildDefnObj = [WebRequestHelper]::InvokeGetWebRequest($build.url);
+                        $repositoryName = $buildDefnObj.repository.name;
+                        $repoSTData = $this.RepositorySTDetails.Data | Where-Object { ($_.repoName -eq $repositoryName)};
+                        if($repoSTData){
+                            $this.BuildSTDetails.data+=@([PSCustomObject] @{ buildDefinitionName = $build.name; buildDefinitionID = $build.id; serviceID = $repoSTData.serviceID; projectName = $repoSTData.projectName; projectID = $repoSTData.projectID; orgName = $repoSTData.orgName } )                            
+                        }                        
+                    }
+                }
+                catch{
+
+                }                
+            }        
         }
+        catch {           
+        } 
         $this.ExportObjToJsonFile($this.BuildSTDetails, 'BuildSTData.json');
 
-        $this.ReleaseSTDetails = Get-content $this.ReleaseMappingsFilePath | ConvertFrom-Json
+        $this.ReleaseSTDetails = Get-content $this.ReleaseMappingsFilePath | ConvertFrom-Json 
         if ([Helpers]::CheckMember($this.ReleaseSTDetails, "data") -and ($this.ReleaseSTDetails.data | Measure-Object).Count -gt 0)
         {
             $this.ReleaseSTDetails.data = $this.ReleaseSTDetails.data | where-object {$_.ProjectName -eq $this.ProjectName}
             if (($this.ReleaseSTDetails.data | Measure-Object).Count -gt 0 -and [string]::IsNullOrWhiteSpace($this.ProjectId))
             {
                 $this.ProjectId = $this.ReleaseSTDetails.data[0].projectId
             }
+        }       
+
+        # Get Release-Repo mappings
+        try {                         
+            $releaseObjectListURL = ("https://vsrm.dev.azure.com/{0}/{1}/_apis/release/definitions?api-version=6.0" ) -f $($this.orgName), $this.projectName;    
+            $releaseObjectList = $this.GetBuildReleaseObjects($ReleaseObjectListURL,'Release');
+            $counter =0
+            foreach ($release in $releaseObjectList) {  
```

Comment:
run loop for unmapped releases only same as build

Reply:
done